chaincode/marbles_transfer/marbles_transfer.go

```diff
@@ -198,11 +199,44 @@ func agreeToPrice(ctx contractapi.TransactionContextInterface, marbleID string,
 	return nil
 }
 
-// TODO implement function to verify marble properties
-// For example, Org1 may tell Org2 about the properties and salt.
-// Org2 would want to verify the properties before agreeing to buy.
-// Org2 would call a verify function on his peer.
-// The properties and salt would passed in, get hashed in the chaincode, and compared with the on-chain hash of the marble properties (queried via GetPrivateDataHash).
+// VerifyAssetProperties implement function to verify marble properties
+func (s *SmartContract) VerifyAssetProperties(ctx contractapi.TransactionContextInterface, marbleID string) (bool, error) {
+	transMap, err := ctx.GetStub().GetTransient()
+	if err != nil {
+		return false, fmt.Errorf("Error getting transient: " + err.Error())
+	}
+
+	// Marble properties are private, therefore they get passed in transient field
+	immutablePropertiesJSON, ok := transMap["marble_properties"]
+	if !ok {
+		return false, fmt.Errorf("marble_properties key not found in the transient map")
+	}
+
+	marble, err := s.GetAsset(ctx, marbleID)
+	if err != nil {
+		return false, fmt.Errorf("failed to get marble: %s", err.Error())
+	}
+
+	collectionSeller := "_implicit_org_" + marble.OwnerOrg
```

Comment:
It's technically not a sale yet... so let's call the variable collectionOwner.

And let's hide this ugly hand crafted internal implicit collection name behind a utility function that takes the owner org id (throughout). Eventually such a utility function may be included in the shim itself.